surefire-api/src/main/java/org/apache/maven/surefire/api/booter/BaseProviderFactory.java

```diff
@@ -115,7 +118,11 @@ private int getThreadCount()
     @Override
     public RunOrderCalculator getRunOrderCalculator()
     {
-        return new DefaultRunOrderCalculator( runOrderParameters, getThreadCount() );
+        TestListResolver testListResolver = testRequest.getTestListResolver();
```

Comment:
I will try to make a proposal but currently we are before M6 release. The integration test is very good.
The MOJO class should evaluate the string, and if e.g. regex is not supported, throw MOJO failure exception. The string should take parallel path along TestListResolver, because it becomes another information for RunOrder, but we must not use TestListResolver. The TestListResolver has nothing to do ordering and it is not a transfer object for `-Dtest=...`. I will try to do something but not this week.